firebase-query.html

```diff
@@ -375,6 +380,10 @@
               this.syncToMemory(function() {
                 this.splice('data', this.__indexFromKey(key), 1);
               });
+              if (this.data.length === 0) {
```

Comment:
Hm... what if `data` is changed in some property effect? Don't you think we should get the length of `data` before calling `this.splice`?

Reply:
We are just assessing whether we have data or not. Sounds correct to me,